### DISCUSSION TITLE
Arrow keys

### DISCUSSION THREAD

**Discussion by jeff-geoff** [isontheline/pro.webssh.net#424] [Ideas]
Could you add the option to use swipe up, swipe down, swipe left, and swipe right as arrow keys? A la Terminus?

**Comment by isontheline** [comment]
Hello @jeff-geoff 👋

Doesn't the tiny pad fit your needs?
![image](https://user-images.githubusercontent.com/44212923/135768088-04329a09-5eec-4623-9f72-24cee67305fe.png)

**Comment by jeff-geoff** [comment]
Sort of, but because it has to start from such a small button it can be difficult to control. Especially since key repeat can not be disabled. I referenced terminus because it works essentially the same way as the button you mention, but they make the entire terminal screen the “button”.

in other words, the gesture that otherwise functions as “scroll left” is actually “arrow left”. Swiping up is arrow up, no matter where you start. One swipe=one key press.

for things like midnight commander, this is a killer feature. For simply cycling through command line history it is unbelievably intuitive.

**Comment by isontheline** [comment]
Thanks for your feedback 🙏

I will improve WebSSH with these issues :
1. https://github.com/isontheline/pro.webssh.net/issues/58
2. https://github.com/isontheline/pro.webssh.net/issues/59
3. And now yours : https://github.com/isontheline/pro.webssh.net/issues/426

**Reply by jeff-geoff** [reply]
Thank you. Just to be clear, I think these issues are similar but not the same. #59 is requesting to simply move the cursor and #426 is requesting to bind the swipe gesture to a specific key press (I.e. arrow keys)

I’m not sure how they can be reconciled, aside from a configurable option for how the app treats swipe input in the terminal portion of the screen?

**Comment by isontheline** [comment]
It's all clear 😇

I have mentioned #58 and #59 just because I will work on the gestures over the terminal but, you're right they are not same as yours.
